ivc/src/ivc/columns.rs

```diff
@@ -19,230 +231,45 @@ use kimchi_msm::{
 pub const N_BLOCKS: usize = 6;
 
 /// Defines the height of each block in the IVC circuit.
+/// As described in the top-level of this file, the circuit is tiled
+/// vertically, and the height of each block is defined by the actual
+/// application that needs to be run.
 pub fn block_height<const N_COL_TOTAL: usize, const N_CHALS: usize>(block_num: usize) -> usize {
     match block_num {
+        // The first block is used for the decomposition of the inputs.
+        // As we do have 3 inputs (left, right and output), we need 3 times the
+        // total number of columns.
         0 => 3 * N_COL_TOTAL,
-        1 => 6 * N_COL_TOTAL + 2,
+        // The second block is used for hashing the different values.
+        // We do have 2 foreign field elements to hash per commitment.
+        // At the moment, we do split each foreign field element in 150 bits
+        // chunk, therefore we need (2 * 2 * 3) (= 12) hashes per commitment.
+        // FIXME: why an additional 2?
+        // FIXME: use the encoding of the bit of y instead. It would reduce to 6
+        // hashes instead of 12.
+        //   For a EC point (x, y), we can only rely on x and the bit sign of y.
+        //   We decompose x into `x_{0..149}` and `x_{150..255}` (only 255 as
+        //   the foreign field is 255 bits), and we add into the value
+        //   `x_{150..255}` the bit sign of `y`. From there, we can hash the two
+        //   values, which fits in our state of size 3.
+        1 => 12 * N_COL_TOTAL + 2,
```

Comment:
See comment for this change. Was 6 before. Added https://github.com/o1-labs/proof-systems/pull/2368 to verify it is still working for our first real usecase.

Reply:
My comment was wrong. I deleted the commit https://github.com/o1-labs/proof-systems/commit/17f4d5dec8658299a42d18642b2eb9023c0e415f